ydb/docs/en/core/integrations/hibernate.md

```diff
@@ -0,0 +1,237 @@
+| `long`, `java.lang.Long`                                            | `Int64`                     |
+| `short`, `java.lang.Short`                                          | `Int16`                     |
+| `byte`, `java.lang.Byte`, enum with `@Enumerated(EnumType.ORDINAL)` | `Int8`                      |
+| `[]byte`                                                            | `Bytes`  (synonym `String`) |
+| `java.time.LocalDateTime`                                           | `Datetime`                  |
+| `java.time.Instant`                                                 | `Timestamp`                 |
+
+Database schema generation based on hibernate entities is supported.
+
+For example, for the Group class:
+
+```java
+@Getter
+@Setter
+@Entity
+@Table(name = "Groups", indexes = @Index(name = "group_name_index", columnList = "GroupName"))
+@NamedQuery(
+        name = "Group.findGroups",
+        query = "SELECT g FROM Group g " +
+                "JOIN Plan p ON g.id = p.planId.groupId " +
+                "JOIN Lecturer l ON p.planId.lecturerId = l.id " +
+                "WHERE p.planId.courseId = :CourseId and l.id = :LecturerId"
+)
+public class Group {
+
+    @Id
+    @Column(name = "GroupId")
+    private int id;
+  
+    @Column(name = "GroupName")
+    private String name;
+  
+    @OneToMany(mappedBy = "group")
+    private List<Student> students;
+  
+    @Override
+    public int hashCode() {
+      return id;
+    }
+}
+```
+
+The following `Groups` table will be created and a secondary index, `group_name_index`, will be added to the `GroupName` column:
+
+```sql
+create table Groups (
+    GroupId Int32 not null,
+    GroupName Text,
+    primary key (GroupId)
+);
+
+alter table Groups
+  add index group_name_index global 
+       on (GroupName);
+```
+
+If you evolve the Group entity by adding the `deparment` field:
+
+```java
+@Column
+private String department;
+```
+
+Hibernate, at the start of the application, update the database schema if the `update` mode is set:
+
+```properties
+jakarta.persistence.schema-generation.database.action=update
+```
+
+The result of changing the scheme is:
+
+```sql
+alter table Groups 
+   add column department Text
+```
+
+{% note warning %}
+
+Hibernate is not designed to manage database schemas.
+
+{% endnote %}
+
+{% note info %}
+
+You can use [Liquibase](./liquibase.md) or Flyway to manage your database schema.
+
+{% endnote %}
+
+### Example with Spring Data JPA
+
+Configure [Spring Data JPA](https://spring.io/projects/spring-data-jpa/) with Hibernate to use custom {{ ydb-short-name }} dialect by updating your `application.properties`:
+
+```properties
+spring.jpa.properties.hibernate.dialect=tech.ydb.hibernate.dialect.YdbDialect
+
+spring.datasource.driver-class-name=tech.ydb.jdbc.YdbDriver
+spring.datasource.url=jdbc:ydb:<grpc/grpcs>://<host>:<2135/2136>/path/to/database[?saFile=file:~/sa_key.json]
+```
+
+Create simple entity and repository:
+
+```kotlin
```

Comment:
Why does it switch from Java to Kotlin here? Ideally, we could provide all examples for both languages in tabs.

Scala would be nice, too, but it doesn't have to be part of this PR.

Reply:
@KirillKurdyukov I still don't understand this

Reply:
I chose Kotlin as an examples